salt/client/__init__.py

```diff
@@ -188,15 +188,15 @@ def __read_master_key(self):
             key_user = key_user.replace('\\', '_')
         keyfile = os.path.join(self.opts['cachedir'],
                                '.{0}_key'.format(key_user))
-        # Make sure all key parent directories are accessible
```

Comment:
@cachedout What do you think about moving this in the try/except below? I wanted to get your opinion on this. Otherwise this PR looks good to me.